### ISSUE TITLE
Stackoverflow in `ceil(Any, 1.0)`

### ISSUE THREAD

**Issue by mschauer** [JuliaLang/julia#33025]
```
julia> ceil(Any, 1.0)
ERROR: StackOverflowError:
Stacktrace:
 [1] ceil(::Type{Any}, ::Float64) at ./missing.jl:116 (repeats 80000 times)
```

**Comment by fredrikekre** [comment]
Fixed by #31602

**Comment by mschauer** [comment]
Great!
